_blog/deploying-laravel-to-kubernetes/index.md

```diff
@@ -0,0 +1,354 @@
+Kubernetes may intimidate most of us, but let's not be. These are 
+simply words to compartmentalise different layers and attributes within a 
+deployment scenario.
+ 
+I have kept this example quite simple by removing state, i.e. no persistence 
+and thus no databases. In this example, we will use Minikube to locally deploy
+ a basic Laravel application across three nodes for load balancing.
+ 
+_Want to learn how to manage storage volumes and databases? We will be 
+writing a tutorial about this soon. Subscribe and be notified as
+ soon as new articles are published._
+ 
+__The application__
+
+I have provided a simple Laravel application which can be cloned from <a href="https://github.com/learnk8s/laravel-kubernetes-demo" target="_blank" rel="noopener" class="link">GitHub</a>. This is nothing more than a new 
+Laravel installation. Therefore you can follow this example using either the 
+demo application or simply create your own new Laravel application. To use 
+the demo application simply clone in your project's directory.
+
+```bash
+cd /to/your/working/directory
+git clone git@github.com:learnk8s/laravel-kubernetes-demo.git .
+```
+
+__Prerequisites__
+
+In order to follow with this demonstration, you will need the following 
+installed on your local system:
+
+1) <a href="https://docs.docker.com/install/" target="_blank" 
+rel="noopener">Docker</a>
+
+2) <a href="https://kubernetes.io/docs/tasks/tools/install-kubectl/" target="_blank" 
+rel="noopener">Kubectl</a>
+
+3) <a href="https://github.com/kubernetes/minikube/releases" target="_blank" 
+   rel="noopener">Minikube</a>
+
+_Having problems installing and running these applications on Windows? We will
+ be writing a tutorial on how to accomplish this very soon. Subscribe and be 
+notified as soon as new articles are published._
+
+__Docker image__
+
+Since this tutorial will be run locally, we can simply build a local Docker 
+Image from the `DockerFile` included in the example code. This `DockerFile` 
+is fairly basic:
+
+```bash
+FROM php:7
+RUN apt-get update -y && apt-get install -y openssl zip unzip git
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+RUN docker-php-ext-install pdo mbstring
+WORKDIR /app
+COPY . /app
+RUN composer install
+
+CMD php artisan key:generate
+CMD php artisan serve --host=0.0.0.0 --port=8181
+EXPOSE 8181
+``` 
+
+It extends a PHP base image, installs some system dependencies including 
+composer and the standard PHP extensions required by Laravel. It will then
+ copy the application files to a working directory and install the application's 
+  dependencies via composer. It also runs a couple of `artisan` commands, 
+  last of which serves the application using the built-in PHP web server. At the
+   end, it exposes port `8181` to the deployment.
+  
+To build the local Docker image:
+
+```bash
+cd /to/your/project/directory
+docker build -t yourname/laravel-kubernetes-demo .
```

Comment:
I assume in this scenario that you're either using the Docker daemon exposed by minikube or the internal docker registry exposed by minikube as an add-on.

If the former is correct, we're missing a step. To build a docker image in minikube, you need to connect the docker client to the remote docker daemon inside Minikube. On a mac that is: `eval $(minikube docker-env)`

For the latter, you need to enable the add-on `minikube add-on registry`. I think. I've never used this one.

Reply:
You're correct. I will fix this. The reason I missed it is that setting the env variable is not needed on Windows. It's also not possible to set it via the CLI.

Reply:
If you add the windows version of the above, I'm happy to include the Linux/MacOSX instructions too.

Reply:
What I'll do is say that if on a UNIX system (where env vars can be set), then the `eval $(minikube docker-env)` step should be added. Otherwise, it is not needed.